File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int, int> m1;
        for(int i=0; i<n; i++){
            m1[nums[i]]++;
        }

        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> heap;
        for(auto &p:m1){
            heap.push({p.second, p.first});
            if(heap.size()>k){
                heap.pop();
            }
        }

        vector<int> res;
        for(int i=0; i<k; i++){
            res.push_back(heap.top().second);
            heap.pop();
        }

        return res;
    }
};

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)
        lo, hi = 0, n-1

        while lo <= hi:
            mid = lo + (hi-lo)//2

            if nums[mid] == target:
                return mid

            print(lo, hi)
            if nums[mid] >= nums[0]:
                if (target < nums[0] or target > nums[mid]):
                    lo = mid + 1
                else:
                    hi = mid - 1

            else:
                if (target >= nums[0] or target < nums[mid]):
                    hi = mid - 1
                else:
                    lo = mid + 1
            
        
        return -1